From c6e1809ae3d0be896d052b255c3c0f5c0d8c0bb4 Mon Sep 17 00:00:00 2001
From: Mayank Dandwani <mayankkk@google.com>
Date: Tue, 4 Mar 2025 12:02:15 -0800
Subject: [PATCH 04/10] Allow core uids to register receiver as "android".

System components can run in any processes belonging to core uids. If
one of such components end up registering a receiver, it will get
registered as "android".

Bug: 387930030
Bug: 310632322
Test: builds
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:252bdb73a318ceccc7391106271b7f233f165810)
Merged-In: Ia577940331af5a79804aa6afcd1773e207a58558
Change-Id: Ia577940331af5a79804aa6afcd1773e207a58558
---
 .../core/java/com/android/server/am/ActivityManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 5d2c13822cd28..e8d9678e6bfec 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -12483,7 +12483,7 @@ public class ActivityManagerService extends IActivityManager.Stub
                             + " (pid=" + Binder.getCallingPid()
                             + ") when registering receiver " + receiver);
                 }
-                if (callerApp.info.uid != SYSTEM_UID
+                if (!UserHandle.isCore(callerApp.info.uid)
                         && !callerApp.getPkgList().containsKey(callerPackage)) {
                     throw new SecurityException("Given caller package " + callerPackage
                             + " is not running in process " + callerApp);
-- 
2.40.1

